💡Add notes on testing, translations, merging and starting Refrisk (#11458)
status-im/status-react: doc/codebase-structure-and-guidelines.md

These guidelines make db.cljs namespaces the place to go when making changes to the db layout and minimize breaking changes when adding/refactoring features

> ## Reframe

Reviewer: ```suggestion
## re-frame
```

- After the PR gets at least 1 approval, it should be moved to "E2E Tests" column. Some PRs may need approvals from more than one person.
- Critical path tests are automatically run for all PRs in "E2E Tests" column.
- If E2E tests pass:
  - If PR doesn't need manual QA (it is small change or change doesn't relate to app) - after passing e2e tests it can be merged (don't use the Github UI to merge, check section on merging PRs).

  - If manual QA needed - add label 'request manual qa'. The QA team will process it (make sure the summary and test instructions are clear). QA team will review results and test PR on different platforms and assign one of the three labels:
    - No issues - label 'TESTED-OK' and then you can merge it.
    - If issues - label 'TESTED-ISSUES' and comment from QA with bugs.
    - After fix and/or discussion the process is repeated.

  - If manual QA is not needed but all tests don't pass, you can ping @churik or @Serhy to confirm that failed E2E tests are not unrelated. 

## Enabling debug logs
Calls to `log/debug` will not be printed to the console by default. It can be enabled under "Advanced settings" in the app:

> ![Enable Debug Logs](https://media.discordapp.net/attachments/626738033701683200/778912893105995786/unknown.png)

Reviewer: not sure this link will be always available

Author: Uploaded image to this repo and created a local link.

## Translations
The app relies on system locale to select a language from the [list of supported languages](https://github.com/status-im/status-react/blob/bda73867471cf2bb8a68b1cc27c9f94b92d9a58b/src/status_im/i18n_resources.cljs#L9). It falls back to English in cash the system locale is not supported.

We use Lokalise App to manage [translations](https://translate.status.im/). In case you need to add/remove a key to translations, you only need to change `en.json`. Missing keys fallback to `en.json`. The actual translations will be added by Lokalise.

> ## Refrisk

Reviewer: ```suggestion
## re-frisk
```

> Refrisk is a state visualization tool written by our very own Andrey (@flexsurfer). To start Refrisk, execute the following command:

Reviewer: ```suggestion
re-frisk is a state visualization tool written by our very own Andrey (@flexsurfer). To start re-frisk, execute the following command:
```

```bash
$ yarn shadow-cljs run re-frisk-remote.core/start

> ```

Reviewer: or 
```bash
$ make run-re-frisk
```
